File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/PermissionStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class PermissionStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "permissionId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private PermissionStateEventId value;

    public PermissionStateEventIdFlattenedDto()
    {
        this(new PermissionStateEventId());
    }

    public PermissionStateEventIdFlattenedDto(PermissionStateEventId value)
    {
        this.value = value;
    }

    public PermissionStateEventId toPermissionStateEventId()
    {
        return this.value;
    }


    public String getPermissionId()
    {
        return this.value.getPermissionId();
    }

    public void setPermissionId(String permissionId)
    {
        this.value.setPermissionId(permissionId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        PermissionStateEventIdFlattenedDto other = (PermissionStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/PermissionStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class PermissionStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'permissionId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var PermissionStateEventId
     */
    private $value;

    /**
     * @param PermissionStateEventId $value
     */
    public function __construct(PermissionStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new PermissionStateEventId();
        }
    }

    /**
     * @return PermissionStateEventId
     */
    public function toPermissionStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getPermissionId()
    {
        return $this->value->getPermissionId();
    }

    /**
     * @param string $permissionId
     */
    public function setPermissionId($permissionId)
    {
        $this->value->setPermissionId($permissionId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getPermissionId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return PermissionStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setPermissionId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

